### PR TITLE
Fixed queue

### PR DESCRIPTION
**Pull request by subashsn**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -11,21 +10,26 @@ class KEAFacility extends BaseFacility {
     this._hasConf = true
     this.leases = []
     super.init()
-    this.fetchServerConf()
   }
 
-  sendCommand (command, service, args = undefined) {
+  setNetFac (netFac) {
+    this.netFac = netFac
+    this.fetchConf()
```
Comment (vigan-abd) [inline index.js]: hmm not sure why these two are connected, shouldn't setNetFac simply set the http client?

Comment (vigan-abd) [reply index.js]: I think you misunderstood, so setNetFac is ok but this.fetchConf didn't make much sense there, now this.netFac is not set anywhere

Comment (subashsn) [reply index.js]: Hey, ya removed the function and doing `this.kea_k0.netFac = this.http_c0` on the worker. Didn't see the need for the function after splitting the logic

File: index.js [hunk 2]
```diff
@@ -11,21 +10,21 @@ class KEAFacility extends BaseFacility {
...
+  async sendCommand (command, service, args = undefined) {
     const body = {
       command,
       service
     }
     if (args) {
       body.arguments = args
     }
-    return axios.post(this.conf.url, body)
+    const data = await this.netFac.post(this.conf.url, { body, encoding: 'json' })
```
Comment (prdn) [inline index.js]: not sure from where `netFac` is derived from

File: index.js [hunk 3]
```diff
@@ -11,21 +12,36 @@ class KEAFacility extends BaseFacility {
...
+      await this.fetchLeases()
+    }
+  }
+
+  setNetFac (netFac) {
+    this.netFac = netFac
+  }
+
+  async sendCommand (command, service, args = undefined) {
+    if (!this.netFac || !this.caller[this.netFac]) {
```
Comment (vigan-abd) [inline index.js]: netFac could be part of opts that facility accept, this way we don't need to deal with it
```
this.caller[this.opts.netFac].post
```

**Review by vigan-abd — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (vigan-abd) [reply index.js]: update:
```
 ['fac', 'bfx-facs-http', 'c0', 'c0', { timeout: 30000, debug: false }, 0],
['fac', 'svc-facs-kea', 'k0', 'k0', () => ({ client: this.http_c0 }), 5]
```
```
this.opts.client.post
```

**Review by vigan-abd — APPROVED** [review]

**Review by vigan-abd — APPROVED** [review]

**Pull request merged**